### PR TITLE
[ENGG-1347] : Update auth modal with new screens and add SSO touchpoint

### PR DESCRIPTION
**Pull request by RuntimeTerror10**
Closes issue: 

## 📜 Summary of changes:

* Replaced in-app auth modal screens with onboarding auth screen
* Added email/password flow in new screens for desktop app
* Added request reset password flow
* Added capture SSO interest touchpoint

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: app/src/features/onboarding/components/auth/components/Form/actions.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+      ({ status }) => {
+        handleVerificationResponse(status, null);
+      }
+    )
+    .catch(() => {
+      handleVerificationResponse(false, null);
+    });
+};
+
+export const handleLogoutButtonOnClick = async (appMode: string, isWorkspaceMode: boolean, dispatch: any) => {
```
Comment (nafees87n) [inline app/src/features/onboarding/components/auth/components/Form/actions.ts:175]: This is not used anywhere

File: app/src/features/onboarding/components/auth/components/Form/actions.ts
Comment (nafees87n) [inline app/src/features/onboarding/components/auth/components/Form/actions.ts:1]: This file looks like a duplicate of `app/src/components/authentication/AuthForm/actions/index.js`.
Do we need to keep both?

Comment (RuntimeTerror10) [reply app/src/features/onboarding/components/auth/components/Form/actions.ts:1]: Yes this is a duplicate, after adding all the auth flows to the new auth screens the whole `authentication` dir can be removed. Right now reset password flow is yet to be moved to new screens

File: app/src/features/onboarding/components/auth/components/Form/components/SSOSignInForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      place: window.location.href,
+      email_type: getEmailType(email),
+      domain: getDomainFromEmail(email),
+      source,
+    });
+
+    setIsLoading(true);
+    const captureSSOInterest = httpsCallable(getFunctions(), "auth-captureSSOInterest");
+
+    captureSSOInterest({ email })
```
Comment (nafees87n) [inline app/src/features/onboarding/components/auth/components/Form/components/SSOSignInForm/index.tsx:49]: What if `success:false` is returned?

Comment (RuntimeTerror10) [reply app/src/features/onboarding/components/auth/components/Form/components/SSOSignInForm/index.tsx:49]: The end result in UI is going to be the same for all cases for now even if interest capture is failed.

**Review by nafees87n — APPROVED** [review]

**Pull request merged**